brigade.js

```diff
@@ -3,98 +3,60 @@ const { events, Job, Group } = require("brigadier")
 const projectOrg = "brigadecore"
 const projectName = "brigade-github-app"
 
-const goImg = "golang:1.11"
+const goImg = "quay.io/deis/lightweight-docker-go:v0.6.0"
 const gopath = "/go"
 const localPath = gopath + `/src/github.com/${projectOrg}/${projectName}`;
 
-const images = [
-  "brigade-github-app",
-  "brigade-github-check-run"
-]
-
 const noop = {run: () => {return Promise.resolve()}}
 
-function build(e, project) {
-  // Create a new job to run Go tests
-  var build = new Job(`${projectName}-build`, goImg);
+const releaseTagRegex = /^refs\/tags\/(v[0-9]+(?:\.[0-9]+)*(?:\-.+)?)$/
 
+function test() {
+  // Create a new job to run Go tests
+  var job = new Job("tests", goImg);
+  job.mountPath = localPath;
   // Set a few environment variables.
-  build.env = {
-      "DEST_PATH": localPath,
-      "GOPATH": gopath
+  job.env = {
+      "SKIP_DOCKER": "true"
   };
-
   // Run Go unit tests
-  build.tasks = [
-    // Need to move the source into GOPATH so vendor/ works as desired.
-    `mkdir -p ${localPath}`,
-    `cp -a /src/* ${localPath}`,
-    `cp -a /src/.git ${localPath}`,
-    `cd ${localPath}`,
-    "make bootstrap",
-    "make lint",
-    "make test"
-  ];
-
-  return build
-}
-
-function goDockerBuild(project, tag) {
-  // We build in a separate pod b/c AKS's Docker is too old to do multi-stage builds.
-  const goBuild = new Job(`${projectName}-docker-build`, goImg);
-
-  goBuild.storage.enabled = true;
-  goBuild.env = {
-    "DEST_PATH": localPath,
-    "GOPATH": gopath
-  };
-  goBuild.tasks = [
-    `cd /src && git checkout ${tag}`,
-    `mkdir -p ${localPath}/bin`,
-    `mv /src/* ${localPath}`,
+  job.tasks = [
     `cd ${localPath}`,
-    "make build-docker-bins",
-    // create share and copy binaries, for use by the dockerhubPublish job
-    `mkdir -p /mnt/brigade/share/rootfs`,
-    `cp -a ./rootfs/* /mnt/brigade/share/rootfs/`,
-    "ls -lah /mnt/brigade/share"
+    "make verify-vendored-code lint test"
   ];
-
-  return goBuild;
+  return job
 }
 
-function dockerhubPublish(project, tag) {
-  const publisher = new Job("dockerhub-publish", "docker");
+function buildAndPublishImages(project, version) {
   let dockerRegistry = project.secrets.dockerhubRegistry || "docker.io";
   let dockerOrg = project.secrets.dockerhubOrg || "brigadecore";
-
-  publisher.docker.enabled = true;
-  publisher.storage.enabled = true;
-  publisher.tasks = [
-    "apk add --update --no-cache make",
+  var job = new Job("build-and-publish-images", "docker:stable-dind");
+  job.privileged = true;
+  job.tasks = [
+    "apk add --update --no-cache make git",
+    "dockerd-entrypoint.sh &",
```

Comment:
New learnings for me here re: the use of this dind image.  Is the `sleep 20` necessary/standard after `dockerd-entrypoint.sh` invocation?

Reply:
The `sleep 20` is kinda ugly, but came from our own docs, believe it or not:

https://docs.brigade.sh/topics/scripting/#docker-in-docker

On other projects, I have opted for a "wupiao.sh" script (Wait Until the Port Is Actually Open; this script actually dates back to the very early days of Deis!)

I could do that here if you think it's worth it, but I felt like just following what our own docs said to do was good for now. lmk

Reply:
Ah, fair enough.  Indeed, let's use what we suggest.  Perhaps later we can refresh/upgrade our docs/suggestion and use here.